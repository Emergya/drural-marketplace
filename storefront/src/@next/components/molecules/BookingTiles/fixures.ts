export const resourceCalendarAvailability = [
  {
    date: "2021-12-27",
    available: true,
  },
  {
    date: "2021-12-28",
    available: false,
  },
  {
    date: "2021-12-29",
    available: true,
  },
  {
    date: "2021-12-30",
    available: false,
  },
  {
    date: "2021-12-31",
    available: true,
  },
  {
    date: "2022-01-01",
    available: false,
  },
  {
    date: "2022-01-02",
    available: false,
  },
  {
    date: "2022-01-03",
    available: true,
  },
  {
    date: "2022-01-04",
    available: false,
  },
  {
    date: "2022-01-05",
    available: true,
  },
  {
    date: "2022-01-06",
    available: false,
  },
  {
    date: "2022-01-07",
    available: true,
  },
  {
    date: "2022-01-08",
    available: false,
  },
  {
    date: "2022-01-09",
    available: false,
  },
  {
    date: "2022-01-10",
    available: false,
  },
  {
    date: "2022-01-11",
    available: false,
  },
  {
    date: "2022-01-12",
    available: false,
  },
  {
    date: "2022-01-13",
    available: false,
  },
  {
    date: "2022-01-14",
    available: false,
  },
  {
    date: "2022-01-15",
    available: false,
  },
  {
    date: "2022-01-16",
    available: false,
  },
  {
    date: "2022-01-17",
    available: true,
  },
  {
    date: "2022-01-18",
    available: false,
  },
  {
    date: "2022-01-19",
    available: true,
  },
  {
    date: "2022-01-20",
    available: false,
  },
  {
    date: "2022-01-21",
    available: true,
  },
  {
    date: "2022-01-22",
    available: false,
  },
  {
    date: "2022-01-23",
    available: false,
  },
  {
    date: "2022-01-24",
    available: true,
  },
  {
    date: "2022-01-25",
    available: false,
  },
  {
    date: "2022-01-26",
    available: true,
  },
  {
    date: "2022-01-27",
    available: false,
  },
  {
    date: "2022-01-28",
    available: true,
  },
  {
    date: "2022-01-29",
    available: false,
  },
  {
    date: "2022-01-30",
    available: false,
  },
  {
    date: "2022-01-31",
    available: true,
  },
  {
    date: "2022-02-01",
    available: false,
  },
  {
    date: "2022-02-02",
    available: true,
  },
  {
    date: "2022-02-03",
    available: false,
  },
  {
    date: "2022-02-04",
    available: true,
  },
];

export const resourceAvailabilityByDate = [
  {
    startTime: "01:45:00",
    endTime: "02:15:00",
  },
  {
    startTime: "02:15:00",
    endTime: "02:45:00",
  },
  {
    startTime: "02:45:00",
    endTime: "03:15:00",
  },
  {
    startTime: "07:45:00",
    endTime: "08:15:00",
  },
  {
    startTime: "08:15:00",
    endTime: "08:45:00",
  },
];
