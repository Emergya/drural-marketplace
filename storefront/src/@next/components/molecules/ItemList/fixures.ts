export const collections = [
  {
    __typename: "Collection",
    id: "1",
    name: "Collection 1",
    slug: "collection-1",
    backgroundImage: null,
  },
  {
    __typename: "Collection",
    id: "2",
    name: "Collection 2",
    slug: "collection-2",
    backgroundImage: null,
  },
  {
    __typename: "Collection",
    id: "3",
    name: "Collection 3",
    slug: "collection-3",
    backgroundImage: null,
  },
  {
    __typename: "Collection",
    id: "4",
    name: "Collection 4",
    slug: "collection-4",
    backgroundImage: null,
  },
  {
    __typename: "Collection",
    id: "5",
    name: "Collection 5",
    slug: "collection-5",
    backgroundImage: null,
  },
  {
    __typename: "Collection",
    id: "6",
    name: "Collection 6",
    slug: "collection-6",
    backgroundImage: null,
  },
];
