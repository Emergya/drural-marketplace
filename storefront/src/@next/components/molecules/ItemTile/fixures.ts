export const collection = {
  __typename: "Collection",
  id: "1",
  name: "Collection 1",
  slug: "collection-1",
  backgroundImage: null,
};

export const category = {
  __typename: "Category",
  id: "1a",
  name: "Category 1",
  slug: "category-1",
  iconId: null,
};
