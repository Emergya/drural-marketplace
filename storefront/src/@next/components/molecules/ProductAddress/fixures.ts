import { ProductAddressCountry } from "@drural/sdk";
import { ProductDetails_address } from "@drural/sdk/lib/fragments/gqlTypes/ProductDetails";

export const PRODUCT_ADDRESS: ProductDetails_address = {
  __typename: "ProductAddressType",
  street: "Pascal",
  postalCode: "28034",
  locality: "Madrid",
  region: "Madrid",
  country: ProductAddressCountry.ES,
  latitude: 37.37301,
  longitude: -5.74951,
};
