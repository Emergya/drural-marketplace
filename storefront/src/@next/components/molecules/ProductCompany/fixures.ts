import { ProductDetails_company } from "@drural/sdk/lib/fragments/gqlTypes/ProductDetails";

export const PRODUCT_COMPANY: ProductDetails_company = {
  __typename: "CompanyType",
  id: "Q29tcGFueVR5cGU6MQ==",
  imageUrl: "/media/companies/dRural_-_Marketp_11.png",
  name: "Comany 01",
  publicName: "Comany 01",
  rating: 2,
  products: {
    __typename: "ProductCountableConnection",
    totalCount: 3,
  },
};
