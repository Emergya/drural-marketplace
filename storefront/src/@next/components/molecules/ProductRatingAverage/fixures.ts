import { IProps } from "./types";

export const DEFAULT_PROPS: IProps = {
  rating: 3,
  reviewsCount: 8,
};
