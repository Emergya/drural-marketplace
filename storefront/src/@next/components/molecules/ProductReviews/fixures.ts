import { GetServiceReviews_product } from "@pages/ReviewsPage/gqlTypes/GetServiceReviews";

export const PRODUCT_REVIEWS: Partial<GetServiceReviews_product> = {
  consumedByUser: true,
  rating: 2,
  reviews: {
    totalCount: 3,
    pageInfo: {
      __typename: "PageInfo",
      hasNextPage: false,
      endCursor: "123123123123",
    },
    __typename: "ProductRatingCountableConnection",
    edges: [
      {
        __typename: "ProductRatingCountableEdge",
        node: {
          __typename: "ProductRating",
          id: "123",
          createdByUser: true,
          user: "Pepito Garcia",
          createdAt: "2021-11-25T07:34:22Z",
          rating: 4,
          comment:
            "This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. ",
        },
      },
      {
        __typename: "ProductRatingCountableEdge",
        node: {
          __typename: "ProductRating",
          id: "123",
          createdByUser: true,
          user: "Pepito Garcia",
          createdAt: "2021-11-25T07:34:22Z",
          rating: 4,
          comment:
            "This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. ",
        },
      },
      {
        __typename: "ProductRatingCountableEdge",
        node: {
          __typename: "ProductRating",
          id: "123",
          createdByUser: true,
          user: "Pepito Garcia",
          createdAt: "2021-11-25T07:34:22Z",
          rating: 4,
          comment:
            "This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. ",
        },
      },
    ],
  },
};
