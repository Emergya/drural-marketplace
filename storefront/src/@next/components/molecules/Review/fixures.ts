import { GetServiceReviews_product_reviews_edges_node } from "@pages/ReviewsPage/gqlTypes/GetServiceReviews";

export const PRODUCT_REVIEW: GetServiceReviews_product_reviews_edges_node = {
  __typename: "ProductRating",
  id: "123",
  createdByUser: true,
  user: "Pepito Garcia",
  createdAt: "2021-11-25T07:34:22Z",
  rating: 4,
  comment:
    "This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. ",
};
