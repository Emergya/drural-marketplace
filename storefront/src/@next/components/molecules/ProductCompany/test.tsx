import "jest-styled-components";

import { mount, ReactWrapper } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";

import { dRuralTheme } from "@styles";

import { ProductCompany } from ".";
import { PRODUCT_COMPANY } from "./fixures";

describe("<ProductInfo />", () => {
  // 1. Variables
  let wrapper: ReactWrapper;

  // 2. Wrapper
  beforeEach(() => {
    wrapper = mount(<ProductCompany company={PRODUCT_COMPANY} />);
  });

  // 3. Tests
  it("exists", () => {
    expect(wrapper.exists()).toEqual(true);
  });

  it("contains provided data", () => {
    expect(wrapper.text()).toContain(PRODUCT_COMPANY.publicName);
    expect(wrapper.text()).toContain(PRODUCT_COMPANY.products?.totalCount);
    expect(wrapper.find("img").at(0).prop("src")).toBe(
      PRODUCT_COMPANY.imageUrl
    );
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={dRuralTheme}>
          <ProductCompany company={PRODUCT_COMPANY} />
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
