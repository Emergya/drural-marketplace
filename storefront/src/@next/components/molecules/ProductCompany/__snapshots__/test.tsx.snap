// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductInfo /> renders correctly 1`] = `
.c6 {
  padding-right: 5px;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  padding-left: 12px;
}

.c2 {
  background-color: #C1BEC8;
  border-radius: 100px;
  height: 112px;
  width: 112px;
}

.c0 {
  margin-bottom: 22px;
}

.c4 {
  margin-bottom: 1rem;
}

.c5 {
  font-size: 0.8125rem;
  line-height: 21px;
  margin-bottom: 12px;
}

@media (max-width:540px) {
  .c0 {
    font-size: 1rem;
    line-height: 1.5rem;
    margin-bottom: 1rem;
  }
}

@media (max-width:540px) {
  .c4 {
    font-size: 18px;
    line-height: 1.5rem;
    margin-bottom: 6px;
  }
}

<div
  className=""
>
  <h4
    className="c0"
  >
    Service provided by
  </h4>
  <div
    className="c1"
  >
    <img
      className="c2"
      src="/media/companies/dRural_-_Marketp_11.png"
    />
    <div
      className="c3"
    >
      <h4
        className="c4"
      >
        Comany 01
      </h4>
      <p
        className="c5"
      >
        3 services published
      </p>
      <span
        style={
          Object {
            "direction": "ltr",
            "display": "inline-block",
          }
        }
      >
        <span
          onClick={[Function]}
          onMouseMove={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          style={
            Object {
              "cursor": "inherit",
              "display": "inline-block",
              "position": "relative",
            }
          }
        >
          <span
            style={
              Object {
                "visibility": "hidden",
              }
            }
          >
            <img
              alt="star-empty"
              className="c6"
              src={[Function]}
            />
          </span>
          <span
            style={
              Object {
                "display": "inline-block",
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
          >
            <img
              alt="star-full"
              className="c6"
              src={[Function]}
            />
          </span>
        </span>
        <span
          onClick={[Function]}
          onMouseMove={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          style={
            Object {
              "cursor": "inherit",
              "display": "inline-block",
              "position": "relative",
            }
          }
        >
          <span
            style={
              Object {
                "visibility": "hidden",
              }
            }
          >
            <img
              alt="star-empty"
              className="c6"
              src={[Function]}
            />
          </span>
          <span
            style={
              Object {
                "display": "inline-block",
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
          >
            <img
              alt="star-full"
              className="c6"
              src={[Function]}
            />
          </span>
        </span>
        <span
          onClick={[Function]}
          onMouseMove={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          style={
            Object {
              "cursor": "inherit",
              "display": "inline-block",
              "position": "relative",
            }
          }
        >
          <span
            style={
              Object {
                "visibility": "hidden",
              }
            }
          >
            <img
              alt="star-empty"
              className="c6"
              src={[Function]}
            />
          </span>
          <span
            style={
              Object {
                "display": "inline-block",
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
          >
            <img
              alt="star-full"
              className="c6"
              src={[Function]}
            />
          </span>
        </span>
        <span
          onClick={[Function]}
          onMouseMove={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          style={
            Object {
              "cursor": "inherit",
              "display": "inline-block",
              "position": "relative",
            }
          }
        >
          <span
            style={Object {}}
          >
            <img
              alt="star-empty"
              className="c6"
              src={[Function]}
            />
          </span>
          <span
            style={
              Object {
                "display": "inline-block",
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "top": 0,
                "width": "0%",
              }
            }
          >
            <img
              alt="star-full"
              className="c6"
              src={[Function]}
            />
          </span>
        </span>
        <span
          onClick={[Function]}
          onMouseMove={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          style={
            Object {
              "cursor": "inherit",
              "display": "inline-block",
              "position": "relative",
            }
          }
        >
          <span
            style={Object {}}
          >
            <img
              alt="star-empty"
              className="c6"
              src={[Function]}
            />
          </span>
          <span
            style={
              Object {
                "display": "inline-block",
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "top": 0,
                "width": "0%",
              }
            }
          >
            <img
              alt="star-full"
              className="c6"
              src={[Function]}
            />
          </span>
        </span>
      </span>
    </div>
  </div>
</div>
`;
